### PR TITLE
Feature/initial commit

### PR DESCRIPTION
**Pull request by xpositivityx**
Still some tidying up to do but here is most of a first pass at DDB.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,54 @@
-# aws-dynamodb-table
\ No newline at end of file
+# aws-dynamodb-table
```
Comment (coryodaniel) [inline README.md:3]: This is the old README template. Is your mass file up to date or has the new README not been added @WillBeebe to mass?

Comment (WillBeebe) [reply README.md:3]: The README in the CLI is the old one. I'll update it with the one from the bulk update script: https://github.com/massdriver-cloud/bundle-github-repo-manager/blob/main/bulk_edit/README.md.tmpl

Comment (coryodaniel) [reply README.md:3]: Save can you drop the same one in here. I’m assuming the precommit is the old one as well.

Comment (xpositivityx) [reply README.md:3]: It definitely was.

File: src/_artifacts.tf [hunk 2]
```diff
@@ -0,0 +1,59 @@
+resource "massdriver_artifact" "table" {
+  field                = "table"
+  provider_resource_id = aws_dynamodb_table.main.arn
+  type                 = "aws-dynamodb-table"
```
Comment (WillBeebe) [inline src/_artifacts.tf]:
```suggestion
```

File: src/_artifacts.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      }
+    }
+  )
+}
+
+resource "massdriver_artifact" "stream" {
+  count                = var.stream.enabled ? 1 : 0
+  field                = "stream"
+  provider_resource_id = aws_dynamodb_table.main.stream_arn
+  type                 = "aws-dynamodb-stream"
```
Comment (WillBeebe) [inline src/_artifacts.tf]:
```suggestion
```

File: src/validations.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "__comment": "TODO: populate this document to put restrictions on provisioner actions. this comment can optionally be removed or used to explain the specified restrictions.",
+    "do_not_delete": []
```
Comment (WillBeebe) [inline src/validations.json]:
```suggestion
    "do_not_delete": [
      "aws_dynamodb_table.main"
    ]
```

**Review by WillBeebe — COMMENTED** [review]
It's looking good, left a few comments. 👍

File: src/_artifacts.tf [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  field                = "table"
+  provider_resource_id = aws_dynamodb_table.main.arn
+  name                 = "DynamoDb table: ${local.name}"
+  artifact = jsonencode(
+    {
+      data = {
+        infrastructure = {
+          arn = aws_dynamodb_table.main.arn
+        }
+        security = {
```
Comment (WillBeebe) [inline src/_artifacts.tf:11]: I really like our `security` block

**Review by likelymichael — APPROVED** [review]

File: src/monitoring.tf [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      stat        = "Sum"
+
+      dimensions = {
+        TableName = local.name
+      }
+    }
+  }
+}
+
+resource "massdriver_package_alarm" "write_capacity" {
```
Comment (WillBeebe) [inline src/monitoring.tf:118]: 👌

**Review by WillBeebe — APPROVED** [review]
LGTM

File: src/monitoring.tf [hunk 7]
```diff
@@ -0,0 +1,122 @@
+module "alarm_channel" {
+  source      = "github.com/massdriver-cloud/terraform-modules//aws-alarm-channel"
+  md_metadata = var.md_metadata
+}
+
+resource "aws_cloudwatch_metric_alarm" "read_capacity" {
```
Comment (coryodaniel) [inline src/monitoring.tf:6]: Not a blocker, but we should def figure out how to make a metric query field on alarms

**Review by coryodaniel — APPROVED** [review]
Looks good to me Did you say that dynamo does magic KMS Cross region

**Comment by xpositivityx** [comment]
> Looks good to me Did you say that dynamo does magic KMS Cross region

They do. If you want customer managed keys, its also zero downtime to create and use those keys.

**Pull request merged**